### ISSUE TITLE
Constantly Walking Away From Stops

### ISSUE THREAD

**Issue by KakunaMatata** [finndev/PokeBuddy#706]
So I've been using PokeBuddy for the last 2 days and it's been working amazingly, I don't really keep an eye on it, but I check back in every 2-3 hours. Every time I check back in I'm in the middle of nowhere or at least 15-20km away from where I started and near maybe 1-3 Pokestops even though I start in an area with 50+.... I decided to watch the bot from when I started it and even ran the program twice, logging in with my friend's account, and on both "Live Location"'s, both characters just walk North and skip a HUGE chunk of the Pokestops..... Is there any way to fix this or restrict the bot to staying in a certain area ?

I've included pictures of where I start and where I end up, thank you :)
![start](https://cloud.githubusercontent.com/assets/20944281/17546720/72d53c9c-5eb1-11e6-9010-6f0f932921cb.png)
![end](https://cloud.githubusercontent.com/assets/20944281/17546723/74ae4130-5eb1-11e6-8fa1-a6a710320284.png)

**Comment by whoppster** [comment]
Same for me. The bot continues to walk but it never refreshes the pokestops.
Also "hangs" when stopping, never stops walking. Thread out of sync?

**Comment by Thorben24** [comment]
Its looking pretty much that your bot cant find pokestops somehow, thats why he is running all the time straight in the same directen. I had the same cause i botted 2 accounts at the same time with the same ip and computer- Change your ip adress should fix the problem (Relog your router or restart it should give you in few seconds a new ip adress if your provider is using a default dynamic solution)

**Comment by whoppster** [comment]
Hm. This has never been a problem for me before, has Niantic implemented some kind of detection for this or is this due to bot design?

**Comment by Thorben24** [comment]
Its a new detection. But this usually only happen if you bot more than 1 account with the same ip (maybe same ip AND +computer) at the same time.

I fixed the problemy by using 2 proxy tunnels for each PokeBuddy instance / process. Im still on the same device, no problems since 6 hours now.

**Comment by KakunaMatata** [comment]
What proxy tunnels are you using ? Are they free ??
I tried ipconfig /release and ipconfig /renew but I'm not sure if it changed my ip or not and haven't tried again yet.

**Comment by Thorben24** [comment]
Im using the software Proxifier (Paid version, but free should work, too, not sure tho) and an own linux server with 2 ip adresses. I setted up 2 proxy servers. Its badly paid, but just 5 usd / month @vultr. And you should have some basic knowledge about Linux Debian, if not, there are many tutorials at youtube or google, its not that way difficult.

**Comment by KakunaMatata** [comment]
Tried setting it up with multiple proxies but it's not working so I'll just
keep trying with my own ip until it works lol thanks though :)

**Comment by Thorben24** [comment]
Hm okay strange. Maybe the proxy servers you tested had a bad connection or were offline? Not sure tho there can be many options why it isnt working.


If you just use 1 account / 1 pokebuddy at the same time you can use your own ip for sure.

**Comment by vendunk** [comment]
This is starting to occur to me even when running only one instance.

**Comment by whoppster** [comment]
Yes. This happens with only one instance as well. Screenshot from when its happening.

![image](https://cloud.githubusercontent.com/assets/20792249/17549392/421f4b76-5ef0-11e6-9b78-5bd874f07f7e.png)

Its not spinning anything, just walking. Is anyone else using the incubator?

**Comment by whoppster** [comment]
And stopping doesn't stop it from walking:

![image](https://cloud.githubusercontent.com/assets/20792249/17549457/8acab900-5ef0-11e6-98fd-53bb22cd4c43.png)

**Comment by whoppster** [comment]
Test without using the incubator, running time 54 minutes, so far no issues. Does the problem lie within the new feature maybe? Starting another bot, enabling the incubator.

**Comment by KakunaMatata** [comment]
I don't use the incubator myself but I was running it for 10 minutes with
no problems and 10 minutes later I checked and was in the middle of nowhere
again. Tried to log in on Nox and it said the servers were having problems.
Assuming the servers are the main issue at the moment but idk. Run it again
and let me know if it works for you. I'm trying to level up my friend as a
surprise before he comes over today haha.

**Comment by whoppster** [comment]
Pretty sure that it has to do with using the incubator. This screenshot is from the bot using the incubator:

![image](https://cloud.githubusercontent.com/assets/20792249/17551838/a283d034-5efc-11e6-91a4-54a9ff662800.png)

It has stopped spinning and farming pokestops.

**Comment by KakunaMatata** [comment]
Is the one without using an incubator running smoothly ?
What location are you using by the way ?

**Comment by whoppster** [comment]
@KakunaMatata Location is somewhere in NY. The one which is not using the incubator is still running smoothly.

**Comment by KakunaMatata** [comment]
Think that's better than the Santa Monica Pier ?

**Comment by KakunaMatata** [comment]
Plus how fast is your walking speed ?

**Comment by whoppster** [comment]
Walking speed: 7km/h.
Moved away from the pier just before the big change happened.

**Comment by whoppster** [comment]
Both bots running for over an hour now, no issues. Will try to deactivate "use only permanent incubator".

**Comment by KakunaMatata** [comment]
I don't have either incubator options running nor any incubators incubating
lol. Guess I'll have to deal with the soft ban and hop over to NYC to try
that out lol.

**Comment by KakunaMatata** [comment]
Still glitching out -_- 
![3](https://cloud.githubusercontent.com/assets/20944281/17553709/5679127a-5ed5-11e6-9620-d64a8ba72d78.png)

**Comment by Thorben24** [comment]
Strange. Your ip is banned again i think? What happen if you close and open pokebuddy again, no pokestops available?

You have to change your ip adress again, also i prefer you to delete pokebuddy and download it again to go sure there is nothing wrong in the configurations you can't change.

**Comment by KakunaMatata** [comment]
Changed my ip, reinstalled PokeBuddy, changed as little settings as
possible, changed to NYC without getting a soft ban and checked on Nox to
make sure the pokestops/pokemon are working (used 2 and caught 3 pokemon)
which they are. Changed my walking speed to 7/km, no incubator settings
checked or incubators running in-game.
Starting my test now.

**Comment by whoppster** [comment]
Both bots are now malfunctioning. Pretty sure im not banned as restarting the bot solves the issue straight away.

**Comment by KakunaMatata** [comment]
I definitely think it's a server issue as the servers keep messing up for a
lot of people including my friends. They'll be down for 10 mins,  then
you'll log in and there's no pokestops or pokemon for 10 mins, then
everything is fine for an hour and it just repeats -_-. I had them running
perfectly fine for almost 3 days and only these past 6 hours have I had
issues including server-sided ones. If it messes up again I'm just gonna
stop for the day and wait till tonight to try again.

**Comment by whoppster** [comment]
Hm. Both bots are in New York. Moving one of them to Miami Beach. None of the bots uses incubators and one of them stopped working after 10 minutes.

**Comment by whoppster** [comment]
If it was a server issue then the GUI would react to me un-checking, for example, "Farm pokestops", right? The server doesn't have anything to do with that, that's just me telling the bot what to do. On the other hand, that could be a second issue which shows its ugly face because of the first...

**Comment by KakunaMatata** [comment]
I've tried again for the past 2 days and it still works for about 10 minutes and then starts going wonky -_- 
This makes no sense and is really annoying.

**Comment by KakunaMatata** [comment]
Just attempted to use my Father's laptop to run the bot and it's been running for 10 minutes perfectly smoothly, will update again in an hour but it must be an ip ban problem. If anyone knows any free VPN programs that work well let me know, thank you.
